samcli/lib/providers/sam_stack_provider.py

```diff
@@ -163,8 +163,14 @@ def _convert_cfn_stack_resource(
         resource_properties: Dict,
         global_parameter_overrides: Optional[Dict] = None,
     ) -> Optional[Stack]:
-        template_url = resource_properties.get("TemplateURL", "")
+        template_url = resource_properties.get("TemplateURL")
 
+        if isinstance(template_url, dict):
+            # This happens when TemplateURL has unresolvable intrinsic functions
+            # and it usually happens in CDK generated template files (#2832).
```

Comment:
wondering if we can show a message to tell customers to try `sam-beta-cdk` ([link](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-cdk.html)

Reply:
Would that solve this problem?